### PR TITLE
[Concept Entry] AI Large Language Models

### PR DESCRIPTION
**Pull request by Hassan-fayez**
### Description

add file that describe the Large Language Models (LLM)
Closes issue #2108 
### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [ ] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by SSwiniarski** [comment]
@Hassan-fayez you seem to have used a e-mail to commit this PR that wasn't connected to your GitHub account. Can you add that e-mail to your account so the CLA knows you signed and you get proper credit when this is merged.

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+Title: 'Large Language Models (LLMs)'
```
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:2]:
```suggestion
Title: 'Large Language Model (LLM)'
```

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:1]: I'd like to rename this to the singular. Changed from `content/ai/concepts/ai-models/models/large-language-models/large-language-models.md` to `content/ai/concepts/ai-models/models/large-language-model/large-language-model.md`

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  - AI
+  - NLP
+  - Deep Learning
+  - Networks
+CatalogContent:
+  - 'introduction-to-large-language-models'
+  - 'how-to-train-a-large-language-model'
+  - 'real-world-applications-of-large-language-models'
+  - 'ethics-and-challenges-of-large-language-models'
+Large Language Models have become increasingly popular in recent years due to their ability to perform a wide range of language-related tasks such as language translation, text summarization, and question-answering. Some examples of Large Language Models include OpenAI's GPT series, Google's BERT, and Facebook's RoBERTa.
```
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:16]:
```suggestion
---

**Large Language Models** are artificial intelligence systems that are designed to process and generate human language on a massive scale. These models are trained on vast amounts of text data and use advanced machine learning algorithms to learn the patterns and structures of language. Large Language Models have become increasingly popular in recent years due to their ability to perform a wide range of language-related tasks such as language translation, text summarization, and question-answering. Some examples of Large Language Models include OpenAI's GPT series, Google's BERT, and Facebook's RoBERTa.
```

Added the ending `---` for the metadata, and also added the description, which should appear in the introduction. (The metadata description is typically a paraphrase of the first sentence(s) of an entry.)

Comment (SSwiniarski) [reply content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:16]: Also, this is one place where off-site links would be appropriate, can you add links to homepages for OpenAI, BERT and RoBERTa?

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+LLMs typically consist of a large neural network architecture that is trained on massive datasets. These models are trained using unsupervised learning techniques, which means that they do not require explicit human supervision to learn. Instead, they learn from the patterns and structures present in the data.
+
+Large Language Models are typically evaluated on their ability to perform a specific task, such as language translation or text generation. They are often fine-tuned on smaller datasets to improve their performance on specific tasks. However, the use of these models has raised concerns about their ethical implications, particularly with regards to bias and the potential for misuse.
+
+Overall, Large Language Models are a powerful tool for processing and generating human language, and their applications are likely to grow in the years to come.
+---
```
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:23]:
```suggestion
```

File: yarn.lock
Comment (SSwiniarski) [inline yarn.lock:1]: Can you undo this change so it is not present in the PR.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@Hassan-fayez I did a first review and left a few comments. One important thing is you need to undo the changes made to the `yarn.lock` file, that shouldn't be touched by a PR.

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+Description: 'Large Language Models are artificial intelligence systems that are designed to process and generate human language on a massive scale. These models are trained on vast amounts of text data and use advanced machine learning algorithms to learn the patterns and structures of language.'
+Subjects:
+  - Machine Learning
+Tags:
+  - AI
+  - NLP
+  - Deep Learning
+  - Networks
+CatalogContent:
+  - 'introduction-to-large-language-models'
```
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:12]: We typically only have two entries for CatalogContent. And one is usually to one of the course paths, an appropriate one to use would be `paths/data-science-nlp` the other entry would be to one of the free courses, `intro-to-chatgpt` would be a good one.

**Review by SSwiniarski — COMMENTED** [review]
A few more comments about the metadata.

**Comment by Hassan-fayez** [comment]
@SSwiniarski 
i have made some changes to the markdown file.
However am still struggling to change my e-mail, will try again later today

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  Tags:
+- AI
+- NLP
+- Deep Learning
+  CatalogContent:
+- 'paths/data-science-nlp'
+- 'intro-to-chatgbt'
+
+---
+
```
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:21]:
```suggestion

**Large Language Models** are artificial intelligence systems that are designed to process and generate human language on a massive scale. These models are trained on vast amounts of text data and use advanced machine learning algorithms to learn the patterns and structures of language. Large Language Models have become increasingly popular in recent years due to their ability to perform a wide range of language-related tasks such as language translation, text summarization, and question-answering. Some examples of Large Language Models include [OpenAI's](https://openai.com) GPT series, Google's BERT, and Facebook's [RoBERTa](https://www.open-roberta.org).

```

This shouldn't be in the description, it's the first paragraph of your entry. Also, I skipped the Wikipedia link, as we don't like linking out to Wikipedia.

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+Title: 'Large Language Models (LLMs)'
+Description: '**Large Language Models** are artificial intelligence systems that are designed to process and generate human language on a massive scale. These models are trained on vast amounts of text data and use advanced machine learning algorithms to learn the patterns and structures of language. Large Language Models have become increasingly popular in recent years due to their ability to perform a wide range of language-related tasks such as language translation, text summarization, and question-answering. Some examples of Large Language Models include OpenAI's GPT series, Google's BERT, and Facebook's RoBERTa.'
```
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:4]:
```suggestion
Description: 'Large Language Models are artificial intelligence systems that are designed to process and generate human language on a massive scale.'
```
The description should be short, one sentence or so. I moved the paragraph back to the start of your entry.

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+homepages:
```
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:5]:
```suggestion
```

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
```
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:6]:
```suggestion
```

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+- https://openai.com
```
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:7]:
```suggestion
```

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+- https://en.wikipedia.org/wiki/BERT_(language_model)
```
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:8]:
```suggestion
```

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+- https://www.open-roberta.org
```
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:9]:
```suggestion
```

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md [hunk 12]
```diff
@@ -0,0 +1,26 @@
+---
+
```
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:2]:
```suggestion
```
Remove extra line break

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+- https://openai.com
+- https://en.wikipedia.org/wiki/BERT_(language_model)
+- https://www.open-roberta.org
+  Subjects:
+- Machine Learning
+  Tags:
+- AI
+- NLP
+- Deep Learning
+  CatalogContent:
```
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:16]:
```suggestion
CatalogContent:
```

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+
+Title: 'Large Language Models (LLMs)'
+Description: '**Large Language Models** are artificial intelligence systems that are designed to process and generate human language on a massive scale. These models are trained on vast amounts of text data and use advanced machine learning algorithms to learn the patterns and structures of language. Large Language Models have become increasingly popular in recent years due to their ability to perform a wide range of language-related tasks such as language translation, text summarization, and question-answering. Some examples of Large Language Models include OpenAI's GPT series, Google's BERT, and Facebook's RoBERTa.'
+homepages:
+
+- https://openai.com
+- https://en.wikipedia.org/wiki/BERT_(language_model)
+- https://www.open-roberta.org
+  Subjects:
+- Machine Learning
```
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:11]:
```suggestion
  - Machine Learning
```

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+  Tags:
+- AI
```
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:13]:
```suggestion
  - AI
```

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+- NLP
```
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:14]:
```suggestion
  - NLP
```

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md [hunk 17]
```diff
@@ -0,0 +1,25 @@
+---
+Title: 'Large Language Models (LLMs)'
+Description: '**Large Language Models** are artificial intelligence systems that are designed to process and generate human language on a massive scale. These models are trained on vast amounts of text data and use advanced machine learning algorithms to learn the patterns and structures of language. Large Language Models have become increasingly popular in recent years due to their ability to perform a wide range of language-related tasks such as language translation, text summarization, and question-answering. Some examples of Large Language Models include [OpenAI's](https://openai.com) GPT series, Google's BERT, and Facebook's [RoBERTa](https://www.open-roberta.org).'
```
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:3]:
```suggestion
Description: 'Large Language Models are artificial intelligence systems that are designed to process and generate human language on a massive scale.'
```
I think you are still misunderstanding. The description should only be one sentence. What you're putting here is what should be the first paragraph of your entry, starting after the metadata. The description doesn't appear in the entry itself. It needs to be short, one sentence or so. You should not put things in the description that don't appear in the entry itself, which starts after the end of the metadata.

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md [hunk 18]
```diff
@@ -0,0 +1,25 @@
...
+  - AI
+  - NLP
+  - Deep Learning
+
+CatalogContent:
+  - 'paths/data-science-nlp'
+  - 'intro-to-chatgpt'
+
+---
+
```
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:18]:
```suggestion

**Large Language Models** are artificial intelligence systems that are designed to process and generate human language on a massive scale. These models are trained on vast amounts of text data and use advanced machine learning algorithms to learn the patterns and structures of language. Large Language Models have become increasingly popular in recent years due to their ability to perform a wide range of language-related tasks such as language translation, text summarization, and question-answering. Some examples of Large Language Models include [OpenAI's](https://openai.com) GPT series, Google's BERT, and Facebook's [RoBERTa](https://www.open-roberta.org).

```
This is where this information should go, not in the description in the metadata.

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md [hunk 19]
```diff
@@ -0,0 +1,25 @@
+---
+Title: 'Large Language Models (LLMs)'
+Description: '**Large Language Models** are artificial intelligence systems that are designed to process and generate human language on a massive scale. These models are trained on vast amounts of text data and use advanced machine learning algorithms to learn the patterns and structures of language. Large Language Models have become increasingly popular in recent years due to their ability to perform a wide range of language-related tasks such as language translation, text summarization, and question-answering. Some examples of Large Language Models include [OpenAI's](https://openai.com) GPT series, Google's BERT, and Facebook's [RoBERTa](https://www.open-roberta.org).'
+
```
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:4]:
```suggestion
```
Remove extra line break.

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md [hunk 20]
```diff
@@ -0,0 +1,25 @@
...
+Subjects:
+  - Machine Learning
+
```
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:7]:
```suggestion
```
Remove extra line break.

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md [hunk 21]
```diff
@@ -0,0 +1,25 @@
...
+Tags:
+  - AI
+  - NLP
+  - Deep Learning
+
```
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:12]:
```suggestion
```
Remove extra line break.

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md [hunk 22]
```diff
@@ -0,0 +1,25 @@
...
+CatalogContent:
+  - 'paths/data-science-nlp'
+  - 'intro-to-chatgpt'
+
```
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:16]:
```suggestion
```
Remove extra line break.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@Hassan-fayez I went through this again, and you are still putting the first paragraph of your entry into the metadata description. This is not where it goes. It belongs in the body of the entry after the metadata. The description should be limited to one sentence.

Also, you need to remove all the extra blank lines from your metadata.

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md (hunk 22 above)
Comment (Hassan-fayez) [reply content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:16]: Sorry for all this issues 😔  it's my first time to use github , I will work on it now.

**Review by SSwiniarski — APPROVED** [review]
@Hassan-fayez Looks ready for a second review.

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md [hunk 23]
```diff
@@ -0,0 +1,21 @@
...
+Tags:
+  - AI
+  - NLP
+  - Deep Learning
+CatalogContent:
+  - 'paths/data-science-nlp'
+  - 'intro-to-chatgpt'
+---
+
+**Large Language Models** are artificial intelligence systems that are designed to process and generate human language on a massive scale. These models are trained on vast amounts of text data and use advanced machine learning algorithms to learn the patterns and structures of language. Large Language Models have become increasingly popular in recent years due to their ability to perform a wide range of language-related tasks such as language translation, text summarization, and question-answering. Some examples of Large Language Models include [OpenAI's](https://openai.com) GPT series, Google's BERT, and Facebook's [RoBERTa](https://www.open-roberta.org).
```
Comment (yangc95) [inline content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:15]:
```suggestion
**Large language models (LLMs)** are artificial intelligence systems that are designed to process and generate human language on a massive scale. These models are trained on vast amounts of text data and use advanced machine learning algorithms to learn the patterns and structures of language. LLMs have become increasingly popular in recent years due to their ability to perform a wide range of language-related tasks such as language translation, text summarization, and question-answering. Some examples of LLMs include [OpenAI's](https://openai.com) GPT series, Google's BERT, and Facebook's [RoBERTa](https://www.open-roberta.org).
```

Would also be worth mentioning bard

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md [hunk 24]
```diff
@@ -0,0 +1,21 @@
...
+
+LLMs typically consist of a large neural network architecture that is trained on massive datasets. These models are trained using unsupervised learning techniques, which means that they do not require explicit human supervision to learn. Instead, they learn from the patterns and structures present in the data.
+
+Large Language Models are typically evaluated on their ability to perform a specific task, such as language translation or text generation. They are often fine-tuned on smaller datasets to improve their performance on specific tasks. However, the use of these models has raised concerns about their ethical implications, particularly with regards to bias and the potential for misuse.
```
Comment (yangc95) [inline content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:19]:
```suggestion
Large language models are typically evaluated on their ability to perform a specific task, such as language translation or text generation. They are often fine-tuned on smaller datasets to improve their performance on specific tasks. However, the use of these models has raised concerns about their ethical implications, particularly with regard to bias and the potential for misuse.
```

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md [hunk 25]
```diff
@@ -0,0 +1,21 @@
...
+
+Overall, Large Language Models are a powerful tool for processing and generating human language, and their applications are likely to grow in the years to come.
```
Comment (yangc95) [inline content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:21]:
```suggestion
Overall, large language models are a powerful tool for processing and generating human language, and their applications are likely to grow in the years to come.
```

File: content/ai/concepts/ai-models/models/large-language-models/large-language-models.md [hunk 26]
```diff
@@ -0,0 +1,21 @@
...
+Tags:
+  - 'AI'
+  - 'NLP'
+  - 'Deep Learning'
+CatalogContent:
+  - 'paths/data-science-nlp'
+  - 'intro-to-chatgpt'
+---
+
+**Large language models (LLMs)** are artificial intelligence systems that are designed to process and generate human language on a massive scale. These models are trained on vast amounts of text data and use advanced machine learning algorithms to learn the patterns and structures of language. LLMs have become increasingly popular in recent years due to their ability to perform a wide range of language-related tasks such as language translation, text summarization, and question-answering. Some examples of LLMs include [OpenAI's](https://openai.com) GPT series, Google's BERT, and Facebook's [RoBERTa](https://www.open-roberta.org).
```
Comment (yangc95) [inline content/ai/concepts/ai-models/models/large-language-models/large-language-models.md:15]:
```suggestion
**Large language models (LLMs)** are artificial intelligence systems that are designed to process and generate human language on a massive scale. These models are trained on vast amounts of text data and use advanced machine learning algorithms to learn the patterns and structures of language. LLMs have become increasingly popular in recent years due to their ability to perform a wide range of language-related tasks such as language translation, text summarization, and question-answering. Some examples of LLMs include [OpenAI's](https://openai.com) GPT series, Google's BERT and Bard (powered by LaMDA), and Facebook's [RoBERTa](https://www.open-roberta.org).
```

**Review by yangc95 — APPROVED** [review]

**Comment by HishamT** [comment]
That sounds like a good idea! 🙂 @yangc95

**Review by HishamT — APPROVED** [review]
LGTM! 👍

**Comment by yangc95** [comment]
Nice @Hassan-fayez live link: https://www.codecademy.com/resources/docs/ai/foundation-models/large-language-models

It looks as if you're anonymous. If you want to be credited, see [My contribution is live but I’m not being shown as a contributor, why is this?](https://discuss.codecademy.com/t/my-contribution-is-live-but-im-not-being-shown-as-a-contributor-why-is-this/758036)

cc: @HishamT @SSwiniarski
